contrib/IECoreUSD/test/IECoreUSD/USDSceneTest.py

```diff
@@ -2489,5 +2489,35 @@ def assertAttributesValues( sphere ):
 		self.assertEqual( readAb.readAttribute( "user:baz", 0 ), IECore.StringData( "white" ) )
 		self.assertEqual( readAb.readAttribute( "render:notUserPrefixAttribute", 0 ), IECore.StringData( "orange" ) )
 
+	def testAttributeBadPrefix( self ):
+		# For Arnold attributes, we currently use a prefix of "ai:" in Cortex, but "arnold:" in USD.
+		# Mixing these up can produce odd results.
+
+		# Using a prefix of "ai:" in USD is completely broken, but shouldn't happen
+		root = IECoreScene.SceneInterface.create( os.path.dirname( __file__ ) + "/data/attributeBadPrefix.usda", IECore.IndexedIO.OpenMode.Read )
+
+		self.assertEqual( set( root.child( "loc" ).attributeNames() ), set( ['INVALID_PREFIX_ai:foo' ] ) )
+		self.assertEqual( root.child( "loc" ).hasAttribute( "INVALID_PREFIX_ai:foo" ), False )
+		self.assertEqual( root.child( "loc" ).hasAttribute( "ai:foo" ), False )
+		self.assertEqual( root.child( "loc" ).hasAttribute( "arnold:foo" ), False )
+		self.assertEqual( root.child( "loc" ).readAttribute( "INVALID_PREFIX_ai:foo", 0 ), None )
+		self.assertEqual( root.child( "loc" ).readAttribute( "ai:foo", 0 ), None )
+		self.assertEqual( root.child( "loc" ).readAttribute( "arnold:foo", 0 ), None )
```

Comment:
So we're reporting the attribute via `attributeNames()`, but then not implementing any of the rest of the API for it? That doesn't seem particularly helpful. It seems like maybe we could make this work if we did a reverse mapping in `findUSDAttribute()`? But if that's true, then couldn't we keep the `ai:` name and have `findUSDAttribute()` fall back to looking for that if it doesn't find `arnold:`?